From 065ae52f4579de089b7919e0e13ed89c2ac94a54 Mon Sep 17 00:00:00 2001
From: randomhydrosol <randomhydrosol@glassrom.org>
Date: Sun, 13 Apr 2025 22:36:40 +0530
Subject: [PATCH 098/101] Add option to set CAKE as default

Intentionally put it on top of the list so that hopefully future changes
to this code are not a problem

Signed-off-by: randomhydrosol <randomhydrosol@glassrom.org>
---
 net/sched/Kconfig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/sched/Kconfig b/net/sched/Kconfig
index 6ddff028b81a4..28a7a61d9810d 100644
--- a/net/sched/Kconfig
+++ b/net/sched/Kconfig
@@ -447,6 +447,9 @@ choice
 	  Select the queueing discipline that will be used by default
 	  for all network devices.
 
+	config DEFAULT_CAKE
+		bool "CAKE" if NET_SCH_CAKE
+
 	config DEFAULT_FQ
 		bool "Fair Queue" if NET_SCH_FQ
 
@@ -468,6 +471,7 @@ endchoice
 
 config DEFAULT_NET_SCH
 	string
+	default "cake" if DEFAULT_CAKE
 	default "pfifo_fast" if DEFAULT_PFIFO_FAST
 	default "fq" if DEFAULT_FQ
 	default "fq_codel" if DEFAULT_FQ_CODEL
-- 
2.51.2

